reference/sql: update compatibility description of MySQL (#1796)
pingcap/docs: dev/reference/mysql-compatibility.md

+ Add Index:
    - Does not support creating multiple indexes at the same time.
    - Does not support the `VISIBLE/INVISIBLE` index.
    - Adding an index on a generated column via `ALTER TABLE` is not supported.

>     - Other Index Type (HASH/BTREE/RTREE) is supported in syntax, rather than in function.

Reviewer: ```suggestion
    - Other Index Type (HASH/BTREE/RTREE) is supported in syntax, but not applicable.
```
The "Other Index Type"s are just ignored and thus "not applicable".

+ `LOCK [=] {DEFAULT|NONE|SHARED|EXCLUSIVE}`: the syntax is supported, but is not applicable to TiDB. All DDL changes that are supported do not lock the table.
+ `ALGORITHM [=] {DEFAULT|INSTANT|INPLACE|COPY}`: the syntax for `ALGORITHM=INSTANT` and `ALGORITHM=INPLACE` is fully supported, but it works differently from MySQL because some operations that are `INPLACE` in MySQL are `INSTANT` in TiDB. The syntax `ALGORITHM=COPY` is not applicable to TIDB and returns a warning.

> Table Option does not support the following syntax:

Reviewer: ```suggestion
The following Table Options are not supported in syntax:
```
How about this?

> The following Table Options are not supported in syntax:

Reviewer 2: ```suggestion
+ The following Table Options are not supported in syntax:
```

    - `WITH/WITHOUT VALIDATION`
    - `SECONDARY_LOAD/SECONDARY_UNLOAD`
    - `CHECK/DROP CHECK`
    - `STATS_AUTO_RECALC/STATS_SAMPLE_PAGES`
    - `SECONDARY_ENGINE`
    - `ENCRYPTION`

> Table Partition does not support the following syntax:

Reviewer: Ditto.

Author: Thanks a lot.

> The following Table Partitions are not supported in syntax:

Reviewer 2: ```suggestion

+ The following Table Partition syntaxes are not supported:
```
The following items are more like "Table Partition syntaxes", not "Table Partitions". Please explain the original Chinese ("Table Partition 不支持以下语法") again if I understand it wrongly. Thanks~ @bb7133

Reviewer: You're right, thanks!
